automation/deploy_igz/deploy_remote.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import logging
+import os
+import re
+import subprocess
+import sys
+from typing import List
+
+import click
+import coloredlogs
+import paramiko
+import yaml
+
+log_level = logging.INFO
+fmt = "%(asctime)s %(levelname)s %(message)s"
+logging.basicConfig(level=log_level)
+logger = logging.getLogger("mlrun-deploy")
+coloredlogs.install(level=log_level, logger=logger, fmt=fmt)
+
+
+class MlRunDeployer(object):
+    class Consts(object):
+        mandatory_fields = ["DATA_NODES", "USER", "PASSWORD", "DOCKER_REGISTRY"]
+
+    def __init__(self, conf_file):
+        self._config = yaml.safe_load(conf_file)
+        for key in self.Consts.mandatory_fields:
+            if self._config.get(key, None) is None:
+                raise RuntimeError(f"Mandatory option {key} not defined")
+        self._username = self._config.get("REGISTRY_USERNAME")
+        self._password = self._config.get("REGISTRY_PASSWORD")
+        if self._username is not None and self._password is None:
+            raise RuntimeError(
+                "REGISTRY_USERNAME defined, yet REGISTRY_PASSWORD is not defined"
+            )
+
+    @staticmethod
+    def _get_image_tag(branch, tag):
+        return f"v{tag}-{branch}"
+
+    @staticmethod
+    def _execute_local_proc_interactive(cmd):
```

Comment:
This isn't a python convention, but it is an Iguazio python convention.
The order of methods in our code bases should be in the order of usage. For example: If method A calls method B, then method A comes before method B. This is so we can read the code in a chronological order.
So please reorder all these helper methods and move them to the end.